lib/seahorn/BvOpSem2FatMemMgr.cc

```diff
@@ -323,96 +389,109 @@ class FatMemManager : public OpSemMemManager {
   FatMemValTy MemSet(PtrTy ptr, Expr _val, unsigned len, FatMemValTy mem,
                      uint32_t align) override {
     return mkFatMem(
-        m_mem.MemSet(mkRawPtr(ptr), _val, len, mkRawMem(mem), align));
+        m_main.MemSet(mkRawPtr(ptr), _val, len, mkRawMem(mem), align),
+        m_main.MemSet(mkRawPtr(ptr), _val, len, mkSlot0Mem(mem), align),
+        m_main.MemSet(mkRawPtr(ptr), _val, len, mkSlot1Mem(mem), align));
   }
 
   /// \brief Executes symbolic memcpy with concrete length
   FatMemValTy MemCpy(PtrTy dPtr, PtrTy sPtr, unsigned len, Expr memTrsfrRead,
                      uint32_t align) override {
-    return mkFatMem(m_mem.MemCpy(mkRawPtr(dPtr), mkRawPtr(sPtr), len,
-                                 mkRawMem(memTrsfrRead), align));
+    return mkFatMem(m_main.MemCpy(mkRawPtr(dPtr), mkRawPtr(sPtr), len,
+                                  mkRawMem(memTrsfrRead), align),
+                    m_slot0.MemCpy(mkRawPtr(dPtr), mkRawPtr(sPtr), len,
+                                   mkSlot0Mem(memTrsfrRead), align),
+                    m_slot1.MemCpy(mkRawPtr(dPtr), mkRawPtr(sPtr), len,
+                                   mkSlot1Mem(memTrsfrRead), align));
   }
 
   /// \brief Executes symbolic memcpy from physical memory with concrete
   /// length
   FatMemValTy MemFill(PtrTy dPtr, char *sPtr, unsigned len, FatMemValTy mem,
                       uint32_t align = 0) override {
     return mkFatMem(
-        m_mem.MemFill(mkRawPtr(dPtr), sPtr, len, mkRawMem(mem), align));
+        m_main.MemFill(mkRawPtr(dPtr), sPtr, len, mkRawMem(mem), align),
+        m_main.MemFill(mkRawPtr(dPtr), sPtr, len, mkSlot0Mem(mem), align),
```

Comment:
same comments as `memset`. Don't need to initialize slots, but if we do, we need to initialize them properly, not with the same values as raw memory

Reply:
still todo

Reply:
memfill funcdoc says it's some kind of memcpy so I copy slot0 and slot1 data also. I don't see why we can let them be uninitialized.

Reply:
`MemFill` takes a concrete memory region and copies it into a symbolic region. The concrete memory region does not have fat part.

Reply:
Done